deployments/engine/templates/scaledobject.yaml

```diff
@@ -0,0 +1,26 @@
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{ include "inference-manager-engine.fullname" . }}-scaledobject
+  labels:
+    {{- include "inference-manager-engine.labels" . | nindent 4 }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ include "inference-manager-engine.fullname" . }}
+  pollingInterval: 15
+  cooldownPeriod: 30
+  minReplicaCount:  {{ .Values.scaledobject.minReplicaCount }}
+  maxReplicaCount:  {{ .Values.scaledobject.maxReplicaCount }}
+  # uncomment this if it is desired to scale to zero.
+  # idleReplicaCount: 0
+  triggers:
+    - type: prometheus
+      metadata:
+        serverAddress: {{ .Values.scaledobject.prometheusServerAddr }}
+        metricName: gpu-util
+        metricType: Value
+        query: |-
+          avg(avg_over_time(DCGM_FI_DEV_GPU_UTIL[1m]))
```

Comment:
Question: Did you allocate GPU to the pod when running this test?

Reply:
Yes, updated deployment spec too